lint/rules.go

```diff
@@ -66,66 +80,19 @@ func NewRuleSet() RuleSet {
 }
 
 func (s *RuleSet) Rules() []Rule {
-	var result []Rule
-	for i := range s.dashboardRules {
-		result = append(result, s.dashboardRules[i])
-	}
-	for i := range s.panelRules {
-		result = append(result, s.panelRules[i])
-	}
-	for i := range s.targetRules {
-		result = append(result, s.targetRules[i])
-	}
-	return result
+	return s.rules
 }
 
-func (s *RuleSet) Add(rule Rule) {
-	if dashboardRule, ok := rule.(DashboardRule); ok {
-		s.dashboardRules = append(s.dashboardRules, dashboardRule)
-	}
-	if panelRule, ok := rule.(PanelRule); ok {
-		s.panelRules = append(s.panelRules, panelRule)
-	}
-	if targetRule, ok := rule.(TargetRule); ok {
-		s.targetRules = append(s.targetRules, targetRule)
-	}
+func (s *RuleSet) Add(r Rule) {
+	s.rules = append(s.rules, r)
 }
 
 func (s *RuleSet) Lint(dashboards []Dashboard) (*ResultSet, error) {
-	resSet := &ResultSet{}
-
-	// Dashboards
-	for _, dashboard := range dashboards {
-		for _, dr := range s.dashboardRules {
-			resSet.AddResult(ResultContext{
-				Result:    dr.LintDashboard(dashboard),
-				Rule:      dr,
-				Dashboard: &dashboard,
-			})
-		}
-		panels := dashboard.GetPanels()
-		for p := range panels {
-			for _, pr := range s.panelRules {
-				resSet.AddResult(ResultContext{
-					Result:    pr.LintPanel(dashboard, panels[p]),
-					Rule:      pr,
-					Dashboard: &dashboard,
-					Panel:     &panels[p],
-				})
-			}
-			targets := panels[p].Targets
-			for t := range targets {
-				for _, tr := range s.targetRules {
-					resSet.AddResult(ResultContext{
-						Result:    tr.LintTarget(dashboard, panels[p], targets[t]),
-						Rule:      tr,
-						Dashboard: &dashboard,
-						Panel:     &panels[p],
-						Target:    &targets[t],
-					})
-				}
-			}
+	resSet := ResultSet{}
```

Comment:
I personally like the resSet := &ResultSet{} syntax and avoid taking pointers on the next lines